src/wallet/wallet.cpp

```diff
@@ -3953,6 +3960,7 @@ CWallet* CWallet::CreateWalletFromFile(const std::string walletFile)
             for (const CWalletTx& wtxOld : vWtx)
             {
                 uint256 hash = wtxOld.GetHash();
+                LOCK(walletInstance->cs_wallet);
```

Comment:
Either:
 - lock once before the loop;
 - loop first with the lock to `mapWallet.find(hash)` for all `vWtx` and then loop again without the lock to `walletdb.WriteTx()`;
 - lock only the mapWallet.find(hash).